eva/udfs/contrib/decorators/ish/io_descriptors/eva_arguments.py

```diff
@@ -0,0 +1,126 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import abstractmethod
+
+
+class EvaArgument(object):
+    """
+    Base class for the data types that are used inside Eva. This class is inherited by the NumPyArray,
+    PyTorchTensor and PandasDataFrame classes.
+    The functions are implemented in the child classes.
+
+    """
+
+    @abstractmethod
+    def __init__(self, shape=None, dtype=None, columns=None) -> None:
```

Comment:
This is unclear to me. Shouldn't each column be it's own EvaArgument? And we also need to have a column name for each of these? Further, we need to be able to pass multiple inputs to a UDF.

Reply:
This should probably be the same as https://github.com/georgia-tech-db/eva/blob/076704705c35245a6c83a626dba599342c59ff64/eva/catalog/models/udf_io_catalog.py#L106